dhcpv6/dhcpv6relay.go

```diff
@@ -180,3 +181,53 @@ func (r *DHCPv6Relay) GetInnerPeerAddr() (net.IP, error) {
 	}
 	return addr, nil
 }
+
+// NewRelayReplFromRelayForw creates a RELAY_REPLY packet based on a RELAY_FORW
+// packet containing the passed DHCPv6 message as payload.
+func NewRelayReplFromRelayForw(relayForw, msg DHCPv6) (DHCPv6, error) {
+	var (
+		err                error
+		linkAddr, peerAddr []net.IP
+		optiids            []Option
+	)
+
+	if relayForw == nil {
+		return nil, errors.New("RELAY_FORW cannot be nil")
+	}
+	if relayForw.Type() != RELAY_FORW {
+		return nil, errors.New("The passed RELAY_FORW must have RELAY_FORW type set")
+	}
+	if msg == nil {
+		return nil, errors.New("The passed message cannot be nil")
+	}
+	if msg.IsRelay() {
+		return nil, errors.New("The passed message cannot be a relay")
+	}
+
+	relay := relayForw.(*DHCPv6Relay)
```

Comment:
I would move this type casting at line 197 together with the check on RELAY_FORW, and check that it's not nil (I know, I'm paranoid)